app/models/timeline_event.rb

```diff
@@ -29,11 +32,13 @@
 class TimelineEvent < ApplicationRecord
   belongs_to :application_form
   belongs_to :creator, polymorphic: true
+  belongs_to :eventable, polymorphic: true, optional: true
 
   enum event_type: {
          assessor_assigned: "assessor_assigned",
          reviewer_assigned: "reviewer_assigned",
-         state_changed: "state_changed"
+         state_changed: "state_changed",
+         assessment_section_completed: "assessment_section_completed"
```

Comment:
I wonder about this name because we've got a `completed` state on assessment sections, I wonder if it will suggest this event is only if the state changed to `completed`. I wonder if something like `assessment_section_recorded` is better?

Reply:
Yeah. I like that.